swiftsimio/writer.py

```diff
@@ -200,6 +201,69 @@ def write_particle_group(self, file_handle: h5py.File, compress: bool):
 
         return
 
+    def write_particle_group_metadata(self, file_handle: h5py.File, dset_attributes):
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            obj = file_handle[f"/PartType{self.particle_type}/{output_handle}"]
+            for attr_name, attr_value in dset_attributes[output_handle].items():
+                obj.attrs.create(attr_name, attr_value)
+
+        return
+
+    def get_attributes(self, scale_factor):
+        attributes_dict = {}
+        a_exp_dict = {
+            "coordinates": 1,
+            "internal_energies": -2,
+        }
+
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            field = getattr(self, name)
+
+            # Find the exponents for each of the dimensions
+            dim_exponents = get_dimensions(field.units.dimensions)
+
+            # Find the scale factor associated quantities
+            a_exp = a_exp_dict.get(name, 0)
+            a_factor = scale_factor * a_exp
+
+            attributes_dict[output_handle] = {
+                "Conversion factor to CGS (not including cosmological corrections)": [
+                    field.unit_quantity.in_cgs()
+                ],
+                "Conversion factor to physical CGS (including cosmological corrections)": [
+                    field.unit_quantity.in_cgs() * a_factor
+                ],
+                "Description": b"Co-moving positions of the particles",
+                "Expression for physical CGS units": b"a U_L  [ cm ]",
```

Comment:
`b"a U_L [cm]"` is not correct for all items. I think we should just leave this as a blank string.

Reply:
ugh, I thought I had removed these... gone now.